{
    "active_lock_reason": null,
    "assignee": null,
    "assignees": [],
    "author_association": "NONE",
    "body": "**Description of the problem**\r\n\r\nI would like to display the number of dots in the legend. This number is not static it comes from a data file.\r\nHowever, I don't understand how to do it with space in the text. The ``label='\"this is my data\"' `` works only for static data.\r\n\r\nMay I help on this ?\r\n\r\n![outputFile](https://user-images.githubusercontent.com/13487830/142998442-c938c29c-22ef-473b-a8d2-d64a2e52297a.png)\r\n\r\n**Full code that generated the error**\r\n\r\nWorks\r\n```python\r\n# Legend labels\r\nlabel_restricted= \"Data_requering_owner`s_authorization_\"+str(data_restricted.size//3)+\"_points\"\r\nlabel_open = \"Available_data_\"+str(data_open.size//3)+\"_points\"\r\n```\r\n\r\nDoes not work\r\n```python\r\n# Legend labels\r\nlabel_restricted= \"Data requering owner`s authorization \"+str(data_restricted.size//3)+\" points\"\r\nlabel_open = \"Available data \"+str(data_open.size//3)+\" points\"\r\n```\r\n\r\nRest of the code\r\n```python\r\n# Create map with shorelines and borders\r\nfig = pygmt.Figure()\r\nfig.basemap(region=region, frame=\"ag\",  projection=\"M25c\")\r\nfig.coast(shorelines=\"1/0.5p,black\", borders=\"1/0.5p,black\")\r\n\r\n# Plot data\r\nif data_restricted.empty == False:\r\n    fig.plot(x=data_restricted.longitude, y=data_restricted.latitude, style=\"t0.08c\", color=\"red\", pen=\"red\", label=f\"{label_restricted}+S0.50c\")\r\n\r\nif data_open.empty == False:\r\n    fig.plot(x=data_open.longitude, y=data_open.latitude, style=\"t0.08c\", color=\"blue\", pen=\"blue\", label=f\"{label_open}+S0.50c\")\r\n\r\n# Add legend outside the map frame\r\nfig.legend(position=\"JMB+o1.4c\", box=True)\r\n```\r\n\r\n**Full error message**\r\n\r\n```\r\n$ python test.py landata_Iles_Baleares.csv\r\nplot [ERROR]: Cannot find file data\r\nplot [ERROR]: Cannot find file 484\r\nplot [ERROR]: Cannot find file points+S0.50c\r\nTraceback (most recent call last):\r\n  File \"/home/tromuald/Documents/BGI/scripts/test.py\", line 67, in <module>\r\n    fig.plot(x=data_open.longitude, y=data_open.latitude, style=\"t0.08c\", color=\"blue\", pen=\"blue\", label=f\"{label_open}+S0.50c\")\r\n  File \"/home/tromuald/anaconda3/envs/pygmt/lib/python3.9/site-packages/pygmt/helpers/decorators.py\", line 805, in new_module\r\n    return module_func(*args, **kwargs)\r\n  File \"/home/tromuald/anaconda3/envs/pygmt/lib/python3.9/site-packages/pygmt/helpers/decorators.py\", line 805, in new_module\r\n    return module_func(*args, **kwargs)\r\n  File \"/home/tromuald/anaconda3/envs/pygmt/lib/python3.9/site-packages/pygmt/helpers/decorators.py\", line 873, in new_module\r\n    return module_func(*args, **kwargs)\r\n  File \"/home/tromuald/anaconda3/envs/pygmt/lib/python3.9/site-packages/pygmt/helpers/decorators.py\", line 586, in new_module\r\n    return module_func(*args, **kwargs)\r\n  File \"/home/tromuald/anaconda3/envs/pygmt/lib/python3.9/site-packages/pygmt/helpers/decorators.py\", line 726, in new_module\r\n    return module_func(*args, **kwargs)\r\n  File \"/home/tromuald/anaconda3/envs/pygmt/lib/python3.9/site-packages/pygmt/src/plot.py\", line 275, in plot\r\n    lib.call_module(\"plot\", arg_str)\r\n  File \"/home/tromuald/anaconda3/envs/pygmt/lib/python3.9/site-packages/pygmt/clib/session.py\", line 500, in call_module\r\n    raise GMTCLibError(\r\npygmt.exceptions.GMTCLibError: Module 'plot' failed with status code 72:\r\nplot [ERROR]: Cannot find file data\r\nplot [ERROR]: Cannot find file 484\r\nplot [ERROR]: Cannot find file points+S0.50c\r\n```\r\n\r\n**System information**\r\n\r\nPlease paste the output of `python -c \"import pygmt; pygmt.show_versions()\"`:\r\n\r\n```\r\n$ python -c \"import pygmt; pygmt.show_versions()\"\r\nPyGMT information:\r\n  version: v0.5.0\r\nSystem information:\r\n  python: 3.9.7 | packaged by conda-forge | (default, Sep 29 2021, 19:20:46)  [GCC 9.4.0]\r\n  executable: /home/tromuald/anaconda3/envs/pygmt/bin/python\r\n  machine: Linux-4.18.0-147.5.1.el8_1.x86_64-x86_64-with-glibc2.28\r\nDependency information:\r\n  numpy: 1.21.4\r\n  pandas: 1.3.4\r\n  xarray: 0.20.1\r\n  netCDF4: 1.5.8\r\n  packaging: 21.0\r\n  ghostscript: 9.54.0\r\n  gmt: 6.2.0\r\nGMT library information:\r\n  binary dir: /home/tromuald/anaconda3/envs/pygmt/bin\r\n  cores: 8\r\n  grid layout: rows\r\n  library path: /home/tromuald/anaconda3/envs/pygmt/lib/libgmt.so\r\n  padding: 2\r\n  plugin dir: /home/tromuald/anaconda3/envs/pygmt/lib/gmt/plugins\r\n  share dir: /home/tromuald/anaconda3/envs/pygmt/share/gmt\r\n  version: 6.2.0\r\n```\r\n",
    "closed_at": "2022-03-13T15:46:29Z",
    "closed_by": {
        "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
        "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
        "followers_url": "https://api.github.com/users/weiji14/followers",
        "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
        "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/weiji14",
        "id": 23487320,
        "login": "weiji14",
        "node_id": "MDQ6VXNlcjIzNDg3MzIw",
        "organizations_url": "https://api.github.com/users/weiji14/orgs",
        "received_events_url": "https://api.github.com/users/weiji14/received_events",
        "repos_url": "https://api.github.com/users/weiji14/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/weiji14"
    },
    "comments": 4,
    "comments_content": [
        {
            "author_association": "NONE",
            "body": "\ud83d\udc4b Thanks for opening your first issue here! Please make sure you filled out the template with as much detail as possible. You might also want to take a look at our [contributing guidelines](https://github.com/GenericMappingTools/pygmt/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/GenericMappingTools/pygmt/blob/main/CODE_OF_CONDUCT.md).\n",
            "created_at": "2021-11-23T09:30:14Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1632#issuecomment-976329203",
            "id": 976329203,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1632",
            "node_id": "IC_kwDOBRZfO846MZnz",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/976329203/reactions"
            },
            "updated_at": "2021-11-23T09:30:14Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/976329203",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/in/4141?v=4",
                "events_url": "https://api.github.com/users/welcome%5Bbot%5D/events{/privacy}",
                "followers_url": "https://api.github.com/users/welcome%5Bbot%5D/followers",
                "following_url": "https://api.github.com/users/welcome%5Bbot%5D/following{/other_user}",
                "gists_url": "https://api.github.com/users/welcome%5Bbot%5D/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/apps/welcome",
                "id": 30606887,
                "login": "welcome[bot]",
                "node_id": "MDM6Qm90MzA2MDY4ODc=",
                "organizations_url": "https://api.github.com/users/welcome%5Bbot%5D/orgs",
                "received_events_url": "https://api.github.com/users/welcome%5Bbot%5D/received_events",
                "repos_url": "https://api.github.com/users/welcome%5Bbot%5D/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/welcome%5Bbot%5D/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/welcome%5Bbot%5D/subscriptions",
                "type": "Bot",
                "url": "https://api.github.com/users/welcome%5Bbot%5D"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "Hi @thomas-romuald, this is a problem related to #247, there are two possible workarounds.\r\n\r\n1. Use `\\040` (the octal code for space) to replace a literal space in your label text as documented in https://github.com/GenericMappingTools/pygmt/issues/247#issuecomment-913890910. So `label=\"this is my data\"` would become `label=\"this\\040is\\040my\\040data\"`.\r\n\r\n2. Use double quotes nested inside single quotes properly. E.g. like so:\r\n```python\r\nimport pygmt\r\n\r\n# Legend labels\r\nlabel_restricted= \"Data requering owner`s authorization \"+str(10//3)+\" points\"\r\nlabel_open = \"Available data \"+str(20//3)+\" points\"\r\nprint(label_restricted)\r\nprint(label_open)\r\n\r\n# Create map with shorelines and borders\r\nfig = pygmt.Figure()\r\nfig.basemap(region=[12, 56, 34, 78], frame=\"ag\",  projection=\"M25c\")\r\nfig.coast(shorelines=\"1/0.5p,black\", borders=\"1/0.5p,black\")\r\n\r\n# Plot data\r\n# if data_restricted.empty == False:\r\nfig.plot(x=12, y=34, style=\"t0.08c\", color=\"red\", pen=\"red\", label=f'\"{label_restricted}\"+S0.50c')\r\n\r\n# if data_open.empty == False:\r\nfig.plot(x=56, y=78, style=\"t0.08c\", color=\"blue\", pen=\"blue\", label=f'\"{label_open}\"+S0.50c')\r\n\r\n# Add legend outside the map frame\r\nfig.legend(position=\"JMB+o1.4c\", box=True)\r\nfig.show()\r\n```\r\n\r\nproduces:\r\n\r\n![image](https://user-images.githubusercontent.com/23487320/143515776-955481c2-98e9-470a-bf3c-f98867d794da.png)\r\n\r\nLet us know if this works for you, or if you have any other questions on usage. We'll try to get this 'space in string' bug fixed in the next version, since it's been an ongoing problem for a while. I'll remind myself to add your failing example as a test in the bugfix at #1487.",
            "created_at": "2021-11-26T02:04:58Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1632#issuecomment-979595962",
            "id": 979595962,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1632",
            "node_id": "IC_kwDOBRZfO846Y3K6",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/979595962/reactions"
            },
            "updated_at": "2021-11-26T02:04:58Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/979595962",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
                "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
                "followers_url": "https://api.github.com/users/weiji14/followers",
                "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
                "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/weiji14",
                "id": 23487320,
                "login": "weiji14",
                "node_id": "MDQ6VXNlcjIzNDg3MzIw",
                "organizations_url": "https://api.github.com/users/weiji14/orgs",
                "received_events_url": "https://api.github.com/users/weiji14/received_events",
                "repos_url": "https://api.github.com/users/weiji14/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/weiji14"
            }
        },
        {
            "author_association": "NONE",
            "body": "Hi @weiji14 thank you very much for your support ! It works now!\r\nAnd thank you for your great work on this tool. I've just started using it recently and I love it!",
            "created_at": "2021-11-26T08:48:50Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1632#issuecomment-979793200",
            "id": 979793200,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1632",
            "node_id": "IC_kwDOBRZfO846ZnUw",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 0,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 0,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/979793200/reactions"
            },
            "updated_at": "2021-11-26T08:48:50Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/979793200",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/13487830?v=4",
                "events_url": "https://api.github.com/users/thomas-romuald/events{/privacy}",
                "followers_url": "https://api.github.com/users/thomas-romuald/followers",
                "following_url": "https://api.github.com/users/thomas-romuald/following{/other_user}",
                "gists_url": "https://api.github.com/users/thomas-romuald/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/thomas-romuald",
                "id": 13487830,
                "login": "thomas-romuald",
                "node_id": "MDQ6VXNlcjEzNDg3ODMw",
                "organizations_url": "https://api.github.com/users/thomas-romuald/orgs",
                "received_events_url": "https://api.github.com/users/thomas-romuald/received_events",
                "repos_url": "https://api.github.com/users/thomas-romuald/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/thomas-romuald/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/thomas-romuald/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/thomas-romuald"
            }
        },
        {
            "author_association": "MEMBER",
            "body": "Cool, great to hear! The team has done an amazing job over the years, and it's always great to see more people using it :grinning:.\r\n\r\nIf you have any other questions, feel free to ask them on the GMT forum at https://forum.generic-mapping-tools.org (you can use your GitHub login to sign in). I'll keep this bug report open for a while until #1487 is merged.",
            "created_at": "2021-11-26T09:58:04Z",
            "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1632#issuecomment-979838950",
            "id": 979838950,
            "issue_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1632",
            "node_id": "IC_kwDOBRZfO846Zyfm",
            "performed_via_github_app": null,
            "reactions": {
                "+1": 1,
                "-1": 0,
                "confused": 0,
                "eyes": 0,
                "heart": 0,
                "hooray": 0,
                "laugh": 0,
                "rocket": 0,
                "total_count": 1,
                "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/979838950/reactions"
            },
            "updated_at": "2021-11-26T09:58:04Z",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/comments/979838950",
            "user": {
                "avatar_url": "https://avatars.githubusercontent.com/u/23487320?v=4",
                "events_url": "https://api.github.com/users/weiji14/events{/privacy}",
                "followers_url": "https://api.github.com/users/weiji14/followers",
                "following_url": "https://api.github.com/users/weiji14/following{/other_user}",
                "gists_url": "https://api.github.com/users/weiji14/gists{/gist_id}",
                "gravatar_id": "",
                "html_url": "https://github.com/weiji14",
                "id": 23487320,
                "login": "weiji14",
                "node_id": "MDQ6VXNlcjIzNDg3MzIw",
                "organizations_url": "https://api.github.com/users/weiji14/orgs",
                "received_events_url": "https://api.github.com/users/weiji14/received_events",
                "repos_url": "https://api.github.com/users/weiji14/repos",
                "site_admin": false,
                "starred_url": "https://api.github.com/users/weiji14/starred{/owner}{/repo}",
                "subscriptions_url": "https://api.github.com/users/weiji14/subscriptions",
                "type": "User",
                "url": "https://api.github.com/users/weiji14"
            }
        }
    ],
    "comments_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1632/comments",
    "created_at": "2021-11-23T09:30:13Z",
    "events_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1632/events",
    "html_url": "https://github.com/GenericMappingTools/pygmt/issues/1632",
    "id": 1061012884,
    "labels": [
        {
            "color": "ee0701",
            "default": true,
            "description": "Something isn't working",
            "id": 563309827,
            "name": "bug",
            "node_id": "MDU6TGFiZWw1NjMzMDk4Mjc=",
            "url": "https://api.github.com/repos/GenericMappingTools/pygmt/labels/bug"
        }
    ],
    "labels_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1632/labels{/name}",
    "locked": false,
    "milestone": {
        "closed_at": "2022-03-15T23:17:52Z",
        "closed_issues": 174,
        "created_at": "2021-06-23T03:22:20Z",
        "creator": {
            "avatar_url": "https://avatars.githubusercontent.com/u/3974108?v=4",
            "events_url": "https://api.github.com/users/seisman/events{/privacy}",
            "followers_url": "https://api.github.com/users/seisman/followers",
            "following_url": "https://api.github.com/users/seisman/following{/other_user}",
            "gists_url": "https://api.github.com/users/seisman/gists{/gist_id}",
            "gravatar_id": "",
            "html_url": "https://github.com/seisman",
            "id": 3974108,
            "login": "seisman",
            "node_id": "MDQ6VXNlcjM5NzQxMDg=",
            "organizations_url": "https://api.github.com/users/seisman/orgs",
            "received_events_url": "https://api.github.com/users/seisman/received_events",
            "repos_url": "https://api.github.com/users/seisman/repos",
            "site_admin": false,
            "starred_url": "https://api.github.com/users/seisman/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/seisman/subscriptions",
            "type": "User",
            "url": "https://api.github.com/users/seisman"
        },
        "description": "",
        "due_on": null,
        "html_url": "https://github.com/GenericMappingTools/pygmt/milestone/9",
        "id": 6880166,
        "labels_url": "https://api.github.com/repos/GenericMappingTools/pygmt/milestones/9/labels",
        "node_id": "MDk6TWlsZXN0b25lNjg4MDE2Ng==",
        "number": 9,
        "open_issues": 0,
        "state": "closed",
        "title": "0.6.0",
        "updated_at": "2022-03-15T23:17:52Z",
        "url": "https://api.github.com/repos/GenericMappingTools/pygmt/milestones/9"
    },
    "node_id": "I_kwDOBRZfO84_PcWU",
    "number": 1632,
    "performed_via_github_app": null,
    "reactions": {
        "+1": 0,
        "-1": 0,
        "confused": 0,
        "eyes": 0,
        "heart": 0,
        "hooray": 0,
        "laugh": 0,
        "rocket": 0,
        "total_count": 0,
        "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1632/reactions"
    },
    "repository_url": "https://api.github.com/repos/GenericMappingTools/pygmt",
    "state": "closed",
    "state_reason": "completed",
    "timeline_url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1632/timeline",
    "title": "Dynamic legend labels with space in it",
    "updated_at": "2022-03-13T15:54:04Z",
    "url": "https://api.github.com/repos/GenericMappingTools/pygmt/issues/1632",
    "user": {
        "avatar_url": "https://avatars.githubusercontent.com/u/13487830?v=4",
        "events_url": "https://api.github.com/users/thomas-romuald/events{/privacy}",
        "followers_url": "https://api.github.com/users/thomas-romuald/followers",
        "following_url": "https://api.github.com/users/thomas-romuald/following{/other_user}",
        "gists_url": "https://api.github.com/users/thomas-romuald/gists{/gist_id}",
        "gravatar_id": "",
        "html_url": "https://github.com/thomas-romuald",
        "id": 13487830,
        "login": "thomas-romuald",
        "node_id": "MDQ6VXNlcjEzNDg3ODMw",
        "organizations_url": "https://api.github.com/users/thomas-romuald/orgs",
        "received_events_url": "https://api.github.com/users/thomas-romuald/received_events",
        "repos_url": "https://api.github.com/users/thomas-romuald/repos",
        "site_admin": false,
        "starred_url": "https://api.github.com/users/thomas-romuald/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/thomas-romuald/subscriptions",
        "type": "User",
        "url": "https://api.github.com/users/thomas-romuald"
    }
}